src/util/prolog/prologStringManipulation.ts

```diff
@@ -1,55 +1,94 @@
+	*/
+export function removeFileExtension(path: string): string {
+	const fetchFileExtension = /(.+)\.[^\.]*/;
 	const fetch: string[] | null = fetchFileExtension.exec(path);
 	if (fetch === null) { return ""; }
 	return fetch[1];
 }
 
-export function removeModuleDeclarations(prologProgram: string) {
+/**
+	* Necessary because module declarations break tau-prolog
+	* @param path - The program code to be curated
+	* @return Program code without module declarations except for library calls
+	*/
+export function removeModuleDeclarations(prologProgram: string): string {
 	const fetchModuleDeclaration = /:-( |)module\([^)]*\) *\.[\r\t\f\v ]*\n?/g;
-
 	return prologProgram.replace(fetchModuleDeclaration, "");
 }
 
-function removeFullLineComments(prologProgram: string) {
+/**
+	* To reduce code
+	* @param path - The program code to be curated
+	* @return Program code without prolog comments that span a whole line including the line
+	*/
+function removeFullLineComments(prologProgram: string): string {
 	const fetchFullLineCommentsRegExp = /^[\r\t\f\v ]*%[^\n]*\n/gm;
-
 	return prologProgram.replace(fetchFullLineCommentsRegExp, "");
 }
 
-function removePartialLineComments(prologProgram: string) {
+/**
+	* To reduce code
+	* @param path - The program code to be curated
+	* @return Program code without prolog comments at the end of line
+	*/
+function removePartialLineComments(prologProgram: string): string {
 	const fetchPartialLineCommentsRegExp = /(^[^\n%]+)%[^\n\r]*(\r?\n)/gm;
-
 	// group2 fetches the end-of-line (EOL) standard in the file to fit both: Windows and UNIX
 	// https://www.networkworld.com/article/3107972/windows-vs-unix-those-pesky-line-terminators.html
 	return prologProgram.replace(fetchPartialLineCommentsRegExp, (_, group1, group2) => group1.trimRight() + group2);
 }
 
-export function removeComments(prologProgram: string) {
-
- return removeFullLineComments(removePartialLineComments(prologProgram));
+/**
+	* To reduce code
+	* @param path - The program code to be curated
+	* @return Program code without prolog comments and the lines of full line comments
+	*/
+export function removeComments(prologProgram: string): string {
+	return removeFullLineComments(removePartialLineComments(prologProgram));
 }
 
-function removeEmptyLines(prologProgram: string) {
+/**
+	* To reduce code
+	* @param path - The program code to be curated
+	* @return Program code without empty lines and lines with only whitespace characters
+	*/
+function removeEmptyLines(prologProgram: string): string {
 	const fetchEmptyLinesRegExp = /\n(?:\s)*\n/gm;
-
 	return prologProgram.replace(fetchEmptyLinesRegExp, "\n");
 }
 
-function removeDoubleWhitespaces(prologProgram: string) {
+/**
+	* To reduce code
+	* @param path - The program code to be curated
+	* @return Program code without double whitespace
+	*/
+function removeDoubleWhitespaces(prologProgram: string): string {
 	const fetchDoubleWhitespacesRegExp = /  */gm;
```

Comment:
This Regex seems to also detect single WhiteSpaces. Is this intended?

Reply:
You're right and that is not intended. Should have been a + instead and that is the case now. The last push made undid your approval though.